### ISSUE TITLE
Input method is not disabled at startup

### ISSUE THREAD

**Issue by senioria** [vim/vim#10513]
### Steps to reproduce

Start vim with IME not disabled defaultly, and vim wouldn't disable it after startup.

### Expected behaviour

IME is disabled when vim starts.

### Version of Vim

8.2.3614

### Environment

OS: Windows
gvim

**Comment by brammool** [comment]
Why is this a problem?  What is the environment?  MS-Windows has various language settings.

**Comment by 91khr** [comment]
> Why is this a problem?

Some people may have their IME enabled by default. Without disabling the IME at startup, they have to disable the IME 
manually at startup if they start at normal-like modes.

> What is the environment?  MS-Windows has various language settings.

My environment is Windows Server 2012, with its default Chinese IME, but it seems that the problem is of Vim, since 
there seems to be no code to disable this at startup.

**Comment by brammool** [comment]
> > Why is this a problem?
> 
> Some people may have their IME enabled by default. Without disabling
> the IME at startup, they have to disable the IME 
> manually at startup if they start at normal-like modes.
> 
> > What is the environment?  MS-Windows has various language settings.
> 
> My environment is Windows Server 2012, with its default Chinese IME,
> but it seems that the problem is of Vim, since 
> there seems to be no code to disable this at startup.

I am not familiar with this setup.  Did you try looking through options?
There is 'iminsert' and 'imsearch', but no 'imnormal'.
Perhaps 'imactivatefunc' can be used to do something.

-- 
BLACK KNIGHT:  I move for no man.
ARTHUR:        So be it!
    [hah] [parry thrust]
    [ARTHUR chops the BLACK KNIGHT's left arm off]
ARTHUR:        Now stand aside, worthy adversary.
BLACK KNIGHT:  'Tis but a scratch.
                                  The Quest for the Holy Grail (Monty Python)

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by k-takata** [comment]
Does it happen with `gvim -u DEFAULTS`?
I'm not familiar with Chinese Windows. At least, Japanse IME is disabled at startup by default.

**Comment by 91khr** [comment]
>Does it happen with `gvim -u DEFAULTS`?

Yes, it does, and it happens even with `gvim -u DEFAULTS -c 'set imdisable'`.

I didn't test with other options, but during editing, the IME switch seems to always be working fine; only at startup 
does the problem happen.

>I'm not familiar with Chinese Windows. At least, Japanse IME is disabled at startup by default.

More interestingly, after installing a Japanese IME, switching to it, and uninstalling it, the deactivation at startup 
also works for Chinese IME...

**Comment by vjeeq** [comment]
thanks #17587 
now I add this config into my vimrc, and it works
```vimscript
au GUIEnter * set imdisable
```
so, maybe vim do something with IME before GUIEnter and IME will do something to start itself (then we see "set imdisable" not work when startup)
BTW, I think vim' s default IME setting is good and I want to disable IME only when startup and then use `set noimdisable`
I use this config
```vimscript
autocmd GUIEnter * set imdisable
autocmd VimEnter * set noimdisable
```
`autocmd GUIEnter * set imdisable` to let IME disabled when vim starts
`autocmd VimEnter * set noimdisable` to use the default setting
